### PR TITLE
Boyscout/building pipeline

### PR DESCRIPTION
**Pull request by RisingLiberty**
Improved the building pipeline.
tools and libs are now versioned.
rexpy is now versioned.
_build.py now takes in project, compiler and config params

### CODE REVIEW

File: build/scripts/setup.py [hunk 1]
```diff
@@ -57,6 +61,7 @@ def run(shouldClean):
 if __name__ == "__main__":
   parser = argparse.ArgumentParser()
   parser.add_argument("-clean", help="clean setup, as if run for the first time", action="store_true")
+  parser.add_argument("-rexpy", help="rexpy installation directory")
```
Comment (RisingLiberty) [inline build/scripts/setup.py:64]: this should be deleted

File: build/sharpmake/src/globals.cs [hunk 2]
```diff
@@ -72,15 +86,15 @@ static public void Init()
...
-    source_root = Path.Combine(root, settings["source_folder"]);
+    source_root = Path.Combine(root, settings.source_folder);
     thirdparty_root = Path.Combine(source_root, "0_thirdparty");
     sharpmake_root = Path.Combine(root, "build", "sharpmake");
-    tools_root = Path.Combine(root, settings["intermediate_folder"], settings["tools_folder"]);
-    libs_root = Path.Combine(root, settings["intermediate_folder"], settings["libs_folder"]);
-    ninja_launcher = Path.Combine(root, settings["ninja_launcher"]);
+
+    // ninja launcher is found in the installation directory of rexpy
+    ninja_launcher = @"c:\users\nickd\appdata\local\programs\python\python310\lib\site-packages\rexpy\ninja_launcher.py"; //Path.Combine(root, settings.ninja_launcher);
```
Comment (RisingLiberty) [inline build/sharpmake/src/globals.cs:96]: should be deleted

File: docs/book.toml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+[book]
+authors = ["Nick De Breuck"]
```
Comment (Dyronix) [inline docs/book.toml:2]: Can you change this to Rex Development Team, we both work on REX

**Review by Dyronix — CHANGES_REQUESTED** [review]
Please remove the rex_std submodule as well

**Comment by Dyronix** [comment]
Could you also make it a bit more clear where the timings point to?

File: _setup.py [hunk 4]
```diff
@@ -13,10 +13,12 @@
 import subprocess
 import sys
 import time
+  
+rexpy_version = "0.0.18"
```
[the remark this answers is not in the document]
Comment (RisingLiberty) [reply _setup.py:17]: it's not finished yet because _test.py doesn't work yet

**Pull request merged**